lib/maze/client/appium/bb_devices.rb

```diff
@@ -18,22 +18,44 @@ def get_available_device(device_or_group_names)
             if device_group_ids
               # Device group found - find a free device in it
               $logger.trace "Got group ids #{device_group_ids} for #{device_or_group_names}"
-              group_count, device = api_client.find_device_in_groups(device_group_ids)
-              if device.nil?
-                raise 'There are no devices available'
+              if device_group_ids.size > 1
+                group_id = false
+                group_count, device = api_client.find_device_in_groups(device_group_ids)
+                if device.nil?
+                  raise 'There are no devices available'
+                else
+                  $logger.info "#{group_count} device(s) currently available in group(s) '#{device_or_group_names}'"
+                end
               else
-                $logger.info "#{group_count} device(s) currently available in group(s) '#{device_or_group_names}'"
+                # Since there is only one group, we can use it verbatim
+                $logger.info "Using device group #{device_or_group_names}"
+                group_id = true
+                device_name = device_group_ids.first
               end
             else
               # See if there is a device with the given name
               device = api_client.find_device device_or_group_names
             end
 
-            device_name = device['displayName']
-            platform = device['platform'].downcase
-            platform_version = device['softwareVersion']['releaseVersion']
+            # If a single device has been identified use that to determine other characteristics
+            if device
+              device_name = device['displayName']
+              platform = device['platform'].downcase
+              platform_version = device['softwareVersion']['releaseVersion']
 
-            $logger.info "Selected device: #{device_name} (#{platform} #{platform_version})"
+              $logger.info "Selected device: #{device_name} (#{platform} #{platform_version})"
+            else
+              # If a device group has been identified, extrapolate characteristics from the group name
+              if android_match = Regexp.new('(ANDROID|android)_(\d{1,2})').match(device_or_group_names)
```

Comment:
I'm not happy with this, as I think it's too inflexible in the event we added a different device group names, however it should be fine for how we access devices at the moment.